zerver/views/user_settings.py

```diff
@@ -192,6 +192,34 @@ def check_settings_values(
         )
 
 
+def check_information_density_setting_values(
+    setting_object: UserProfile | RealmUserDefault,
+    dense_mode: bool | None,
+    web_font_size_px: int | None,
+    web_line_height_percent: int | None,
+) -> None:
+    dense_mode = dense_mode if dense_mode is not None else setting_object.dense_mode
+    web_font_size_px = (
+        web_font_size_px if web_font_size_px is not None else setting_object.web_font_size_px
+    )
+    web_line_height_percent = (
+        web_line_height_percent
+        if web_line_height_percent is not None
+        else setting_object.web_line_height_percent
+    )
+
+    if dense_mode:
+        if web_font_size_px != UserBaseSettings.WEB_FONT_SIZE_PX_LEGACY:
+            raise JsonableError(
+                _("Incompatible values for 'dense_mode' and 'web_font_size_px' settings.")
+            )
+
+        if web_line_height_percent != UserBaseSettings.WEB_LINE_HEIGHT_PERCENT_LEGACY:
+            raise JsonableError(
+                _("Incompatible values for 'dense_mode' and 'web_line_height_percent' settings.")
+            )
```

Comment:
This should probably be done as an `exceptions.py` base class that takes the two parameter names as parameters, to avoid needing to do translations an extra time for every pair of values.

Reply:
`IncompatibleParametersError` looks like it's about the parameter types, not the values, so we probably need a new ``IncompatibleParameterValuesError` for this. Would be good to do ASAP since we'll want to avoid uploading for translators the strings we're removing.